src/v/utils/fragmented_vector.h

```diff
@@ -47,12 +47,13 @@ class fragmented_vector {
     using value_type = T;
 
     fragmented_vector() noexcept = default;
-    fragmented_vector(const fragmented_vector&) noexcept = delete;
     fragmented_vector& operator=(const fragmented_vector&) noexcept = delete;
     fragmented_vector(fragmented_vector&&) noexcept = default;
     fragmented_vector& operator=(fragmented_vector&&) noexcept = default;
     ~fragmented_vector() noexcept = default;
 
+    fragmented_vector copy() const { return *this; }
```

Comment:
can you explain? I don't really understand what's the purpose is. I mean, I get why noexcept can be conditional, but here we've already marked all constructors noexcept unconditionally.

Reply:
I made copy() noexcept

Reply:
>  already marked all constructors noexcept unconditionally.

I guess I was asking if that was intentional. When the copy constructor was deleted, it was noexcept, but then it became implemented, and potentially throwing (making it noexcept is just a way to make the program terminate rather than throw bad_alloc) - was that intentional?

Reply:
ahh. yes it was intentional